src/ui/reducers/timeline.ts

```diff
@@ -42,6 +45,12 @@ const timelineSlice = createSlice({
     setTrimRegion(state, action: PayloadAction<FocusRegion | null>) {
       state.focusRegion = action.payload;
     },
+    pointsReceived(state, action: PayloadAction<TimeStampedPoint[]>) {
+      state.points = sortBy(
```

Comment:
:hand: This _almost_ feels like something we could use an entity adapter for?

If `p.time` is effectively acting like a unique ID, we could do an `adapter.upsertMany(action.payload)`, and also use its `comparator` option to manage the sorting.

Probably not a big difference either way - I'm just pattern-matching on tools I know here.

Reply:
Yeah, part of my reason for suggesting an entity adapter here _was_ to potentially simplify perf behavior, per your comment below :)

but also I was just trying to read this line of code and understand what it's doing, and it looks like it's doing a combo of "unique ID by timestamp" and "keep this sorted", and those are also things that an entity adapter can help with.

Reply:
Fair!